meteor/client/ui/Settings/Studio/Routings.tsx

```diff
@@ -12,723 +11,903 @@ import {
+											/>
 											<EditAttribute
 												modifiedClassName="bghl"
-												attribute={`routeSets.${routeId}.name`}
-												obj={this.props.studio}
+												attribute={`routeSets.${routeSetId}.routes.${index}.remapping.deviceId`}
+												obj={studio}
 												type="text"
 												collection={Studios}
 												className="input text-input input-l"
 											></EditAttribute>
-											<span className="text-s dimmed field-hint">{t('Display name of the Route Set')}</span>
-										</label>
-
-										<label className="field">
-											<LabelActual label={t('Exclusivity group')} />
-											<div>
-												<EditAttribute
-													modifiedClassName="bghl"
-													attribute={`routeSets.${routeId}.exclusivityGroup`}
-													obj={this.props.studio}
-													type="checkbox"
-													className="mrs mvxs"
-													collection={Studios}
-													mutateDisplayValue={(v) => (v === undefined ? false : true)}
-													mutateUpdateValue={() => undefined}
-												/>
-												<EditAttribute
-													modifiedClassName="bghl"
-													attribute={`routeSets.${routeId}.exclusivityGroup`}
-													obj={this.props.studio}
-													type="dropdown"
-													options={Object.keys(this.props.studio.routeSetExclusivityGroups)}
-													mutateDisplayValue={(v) => (v === undefined ? 'None' : v)}
-													collection={Studios}
-													className="input text-input input-l"
-												></EditAttribute>
-											</div>
-											<span className="text-s dimmed field-hint">
-												{t('If set, only one Route Set will be active per exclusivity group')}
-											</span>
-										</label>
-
-										<label className="field">
-											<LabelActual label={t('Behavior')} />
-											<EditAttribute
-												modifiedClassName="bghl"
-												attribute={`routeSets.${routeId}.behavior`}
-												obj={this.props.studio}
-												type="dropdown"
-												options={StudioRouteBehavior}
-												optionsAreNumbers={true}
-												collection={Studios}
-												className="input text-input input-l"
-											></EditAttribute>
-											<span className="text-s dimmed field-hint">
-												{t('The way this Route Set should behave towards the user')}
-											</span>
-										</label>
-									</div>
-									{this.renderRoutes(routeSet, routeId, manifest)}
-									<div className="mod">
-										<button className="btn btn-primary right" onClick={() => this.finishEditItem(routeId)}>
-											<FontAwesomeIcon icon={faCheck} />
-										</button>
-										<button className="btn btn-secondary" onClick={() => this.addNewRouteInSet(routeId)}>
-											<FontAwesomeIcon icon={faPlus} />
-										</button>
-									</div>
-								</td>
-							</tr>
-						)}
-					</React.Fragment>
+										</div>
+									</label>
+
+									<DeviceMappingSettings
```

Comment:
This `DeviceMappingSettings` component also needs replacing, as it is also writing directly to the collection and will no longer work